Feature 651 version set in one place

georgemccabe opened the pull request:
Closes #651 

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>
Added feature branch to Docker build rules and verified that the MET Docker image is selected properly (develop for beta,rc versions, X+6.Y for maj/minor releases)

- [X] Recommend testing for the reviewer to perform, including the location of input datasets:</br>

1) Run master_metplus.py and verify that correct version is shown
2) Check Travis build for Pull Request to see if correct data volumes are used (should all be from develop). Example: https://travis-ci.com/github/dtcenter/METplus/jobs/399913159#L318
From there, click 'Pull Requests' tab at the top, find pull request that corresponds to the feature branch, go to 'Use Case Tests - met_tool_wrapper' under Tests and find equivalent line.

- [X] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [X] After merging, should the reviewer **DELETE** the feature branch from GitHub? **[Yes]**</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [X] After submitting the PR, select **Linked Issues** with the original issue number.

georgemccabe committed:
added __version__ to metplus that reads from docs/version

georgemccabe committed:
read version file, use 'latest' tag of MET Docker image for released versions, i.e. v4.0, use 'develop' tag of MET Docker image for all others, i.e. feature branches, beta releases, etc.

georgemccabe committed:
read version, use release version of data volume from dtcenter/metplus-data if testing a release, i.e. main_vX.Y, otherwise use metplus-data-dev

georgemccabe committed:
fixed call to dirname

georgemccabe committed:
Update conf.py

georgemccabe committed:
[ci-skip] changed version to major release to test docker build hook changes

georgemccabe committed:
Merge branch 'feature_651_version' of github.com:dtcenter/METplus into feature_651_version

georgemccabe committed:
removed v from version check because version file does not include v at the beginning

georgemccabe committed:
changed version back -- setting version to 4.0 successfully used 'latest' for MET image tag

georgemccabe committed:
print version number to screen before anything else is done in master_metplus

georgemccabe committed:
merged develop and resolved conflicts

georgemccabe committed:
set met version to X+6.Y for maj/min release, changed version to test [ci-skip]

georgemccabe committed:
put met version computation into a script that is called by the build hooks file, changed version file back

georgemccabe committed:
removed commented line that should not be needed

georgemccabe committed:
remove duplicate tag that was commented out

georgemccabe committed:
fixed call to get_met_version in build hooks file

georgemccabe committed:
Removed accidentally commented line

georgemccabe commented on ci/travis_jobs/get_data_volumes.py:
```diff
@@ -7,8 +7,20 @@
+# if version is set to vX.Y without -betaZ or -dev, use that version
+# otherwise use develop
+if len(__version__.split('-')) == 1:
+    METPLUS_VERSION = __version__
```
```suggestion
    # only get first 2 numbers from version, i.e. X.Y.Z will use vX.Y
    METPLUS_VERSION = f"v{'.'.join(__version__.split('.')[:2])}"
```

georgemccabe commented on ci/travis_jobs/get_data_volumes.py:
```diff
@@ -7,8 +7,20 @@
+from metplus import __version__
+
+# version should be set to develop or a release version, i.e. vX.Y
+# if version is set to vX.Y without -betaZ or -dev, use that version
```
```suggestion
# if version is set to X.Y without -betaZ or -dev, use that version
```

georgemccabe commented on ci/travis_jobs/get_data_volumes.py:
```diff
@@ -7,8 +7,20 @@
+
+from metplus import __version__
+
+# version should be set to develop or a release version, i.e. vX.Y
```
```suggestion
# METPLUS_VERSION should be set to develop or a release version, i.e. vX.Y
```

georgemccabe committed:
Made comment more clear

[ci-skip]

georgemccabe committed:
Fixed typo in comment

georgemccabe committed:
use develop as version in docs if not a major/minor/bugfix release

JohnHalleyGotway committed:
Update ci/travis_jobs/get_data_volumes.py

Yes, this makes sense. The docker data volumes DO NOT change for bugfix versions. So we should look for docker data volumes tagged as vX.Y, not vX.Y.Z

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

JohnHalleyGotway commented on ci/docker/hooks/get_met_version:
```diff
@@ -0,0 +1,14 @@
+if cat $version_file | egrep -q '[0-9.]+$'; then
+  let major=$(cut -d '.' -f1 $version_file)+6
+  minor=$(cut -d '.' -f2 $version_file )
+  met_branch=$major"."$minor
```
This is where we'd add a reference to the X.Y-latest tag. Correct?
```suggestion
  met_branch=$major"."$minor"-latest"
```

georgemccabe replied:
Correct

JohnHalleyGotway commented on docs/conf.py:
```diff
@@ -13,23 +13,34 @@
+
+verinfo = version
 
 release_year = '2020'
```
Would be nice if there were a way to avoid having to manually update an explicit date. Any ideas?

georgemccabe replied:
You can get the current date easily with Python, but if we had to push updates to an older version, the date would be overwritten. We could add an argument 'release' to the build_docs.py script that is used when creating a release that will get the current date and save it into a text file in the docs directory that will be used if it exists on an future update. This would have to be done after the main_vX.Y branch is created so it is only added to that branch. Also, we would have to add instructions when creating a release to commit that file. Thoughts?

JohnHalleyGotway replied:
Yeah, good points you raise. If manual intervention is required anyway, might as well just keep it as is.

JohnHalleyGotway committed:
Update ci/docker/hooks/get_met_version

Accepting suggestion to switch the MET DockerHub tag from $major"."$minor to $major"."$minor"-latest"
